lessons/pydata/pandas/index.ipynb

```diff
@@ -0,0 +1,2872 @@
+       "      <td>807</td>\n",
+       "      <td>zeraora</td>\n",
+       "      <td>1.5</td>\n",
+       "      <td>44.5</td>\n",
+       "      <td>yellow</td>\n",
+       "      <td>humanoid</td>\n",
+       "      <td>False</td>\n",
+       "      <td>Electric</td>\n",
+       "      <td>NaN</td>\n",
+       "      <td>88</td>\n",
+       "      <td>112</td>\n",
+       "      <td>75</td>\n",
+       "      <td>143</td>\n",
+       "      <td>YELLOW</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "<p>807 rows × 14 columns</p>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "      id         name  height  weight   color      shape  is baby    type 1  \\\n",
+       "0      1    bulbasaur     0.7     6.9   green  quadruped    False     Grass   \n",
+       "1      2      ivysaur     1.0    13.0   green  quadruped    False     Grass   \n",
+       "2      3     venusaur     2.0   100.0   green  quadruped    False     Grass   \n",
+       "3      4   charmander     0.6     8.5     red    upright    False      Fire   \n",
+       "4      5   charmeleon     1.1    19.0     red    upright    False      Fire   \n",
+       "..   ...          ...     ...     ...     ...        ...      ...       ...   \n",
+       "802  803      poipole     0.6     1.8  purple    upright    False    Poison   \n",
+       "803  804    naganadel     3.6   150.0  purple      wings    False    Poison   \n",
+       "804  805    stakataka     5.5   820.0    gray  quadruped    False      Rock   \n",
+       "805  806  blacephalon     1.8    13.0   white   humanoid    False      Fire   \n",
+       "806  807      zeraora     1.5    44.5  yellow   humanoid    False  Electric   \n",
+       "\n",
+       "     type 2  hp  attack  defense  speed color_uppercase  \n",
+       "0    Poison  45      49       49     45           GREEN  \n",
+       "1    Poison  60      62       63     60           GREEN  \n",
+       "2    Poison  80      82       83     80           GREEN  \n",
+       "3       NaN  39      52       43     65             RED  \n",
+       "4       NaN  58      64       58     80             RED  \n",
+       "..      ...  ..     ...      ...    ...             ...  \n",
+       "802     NaN  67      73       67     73          PURPLE  \n",
+       "803  Dragon  73      73       73    121          PURPLE  \n",
+       "804   Steel  61     131      211     13            GRAY  \n",
+       "805   Ghost  53     127       53    107           WHITE  \n",
+       "806     NaN  88     112       75    143          YELLOW  \n",
+       "\n",
+       "[807 rows x 14 columns]"
+      ]
+     },
+     "execution_count": 19,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Velmi mocná je i metoda `apply`, která nám umožní aplikovat jakoukoli funkci napříč celou tabulkou či sloupcem."
```

Comment:
`apply` bych skoro na začátku vynechal - už jsem viděl spoustu lidí kteří pak na vše používají `apply` (nebo `map` ) místo nativních Pandas metod

Reply:
Je to jen ukázka. Pak budeme samozřejmě používat `apply` jen tam, kde to dává smysl.